upup/pkg/fi/cloudup/scaleway/cloud.go

```diff
@@ -16,7 +16,393 @@ limitations under the License.
+
+	nodeMap := cloudinstances.GetNodeMap(nodes, cluster)
+
+	serverGroups, err := findServerGroups(s, cluster.Name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to find server groups: %w", err)
+	}
+
+	for igName, serverGroup := range serverGroups {
+		var instanceGroup *kops.InstanceGroup
+		for _, ig := range instancegroups {
+			if igName == ig.Name {
+				instanceGroup = ig
+				break
+			}
+		}
+		if instanceGroup == nil {
+			if warnUnmatched {
+				klog.Warningf("Server group %q has no corresponding instance group", igName)
+			}
+			continue
+		}
+
+		groups[instanceGroup.Name], err = buildCloudGroup(instanceGroup, serverGroup, nodeMap)
+		if err != nil {
+			return nil, fmt.Errorf("failed to build cloud group for instance group %q: %w", instanceGroup.Name, err)
+		}
+	}
+
+	return groups, nil
+}
+
+func findServerGroups(s *scwCloudImplementation, clusterName string) (map[string][]*instance.Server, error) {
+	servers, err := s.GetClusterServers(clusterName, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	serverGroups := make(map[string][]*instance.Server)
+	for _, server := range servers {
+		igName := ""
+		for _, tag := range server.Tags {
+			if strings.HasPrefix(tag, TagInstanceGroup) {
+				igName = strings.TrimPrefix(tag, TagInstanceGroup+"=")
+				break
+			}
+		}
+		serverGroups[igName] = append(serverGroups[igName], server)
+	}
+
+	return serverGroups, nil
+}
+
+func buildCloudGroup(ig *kops.InstanceGroup, sg []*instance.Server, nodeMap map[string]*v1.Node) (*cloudinstances.CloudInstanceGroup, error) {
+	cloudInstanceGroup := &cloudinstances.CloudInstanceGroup{
+		HumanName:     ig.Name,
+		InstanceGroup: ig,
+		Raw:           sg,
+		MinSize:       int(fi.Int32Value(ig.Spec.MinSize)),
+		TargetSize:    int(fi.Int32Value(ig.Spec.MinSize)),
+		MaxSize:       int(fi.Int32Value(ig.Spec.MaxSize)),
+	}
+
+	for _, server := range sg {
+		status := cloudinstances.CloudInstanceStatusUpToDate
+		cloudInstance, err := cloudInstanceGroup.NewCloudInstance(server.ID, status, nodeMap[server.ID])
+		if err != nil {
+			return nil, fmt.Errorf("failed to create cloud instance for server %s(%s): %w", server.Name, server.ID, err)
+		}
+		cloudInstance.State = cloudinstances.State(server.State)
+		cloudInstance.MachineType = server.CommercialType
+		for _, tag := range server.Tags {
+			if strings.HasPrefix(tag, TagNameRolePrefix) {
+				cloudInstance.Roles = append(cloudInstance.Roles, strings.TrimPrefix(tag, TagNameRolePrefix))
+			}
+		}
+		if server.PrivateIP != nil {
+			cloudInstance.PrivateIP = *server.PrivateIP
+		}
+	}
+
+	return cloudInstanceGroup, nil
+}
+
+func (s *scwCloudImplementation) FindClusterStatus(cluster *kops.Cluster) (*kops.ClusterStatus, error) {
+	klog.V(8).Info("FindClusterStatus is not implemented yet for Scaleway")
+	return nil, nil
+}
+
+func (s *scwCloudImplementation) GetApiIngressStatus(cluster *kops.Cluster) ([]fi.ApiIngressStatus, error) {
+	//TODO(Mia-Cross) implement me
```

Comment:
Why do you panic for `GetApiIngressStatus` and just emit a log message for `FindClusterStatus`?